--- a/tools/cmake/build.cmake	2026-10-09 07:26:28
+++ b/tools/cmake/build.cmake	2026-10-09 07:26:28
@@ -463,6 +463,7 @@
     idf_build_get_property(build_properties __BUILD_PROPERTIES)
     foreach(property ${build_properties})
         idf_build_get_property(val ${property})
+        string(REPLACE "\"" "\\\"" val "${val}")
         set(build_properties_text "${build_properties_text}\nset(${property} \"${val}\")")
     endforeach()
     file(WRITE ${output_file} "${build_properties_text}")
